### PR TITLE
Dsd 1766/npm package update

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-1766](https://jira.nypl.org/browse/DSD-1766)

## This PR does the following:

The original intention for this ticket was to remove and update some packages that were causing security issues, namely `tough-cookie` and `vite`. This then turned into updating Storybook to `8.0.8` and related packages. 

- Adds the following npm packages: `jest-transformer-svg`, `@storybook/addon-designs`, and `@storybook/test`.
- Updates the following packages:
  - Storybook main package and related libraries to `8.0.0`.
  - `remark-gfm` to `4.0.0`.
  - `vite` to `5.2.8`, `@vitejs/plugin-react` to `4.2.1`, and `vite-plugin-svgr` to `4.2.0`.
- Updates how SVGs are imported into `LogoSvgs.tsx` and `IconSvgs.tsx`.
- Updates Storybook internal link and heading styles and enables the hash heading URL linking.
- Removes the following packages: `downshift`, `tough-cookie`, `@storybook/testing-library`, `storybook-addon-designs`, and `@svgr/webpack`.

## How has this been tested?

To test this locally, remove `node_modules` and `package-lock.json` and reinstall packages.

Storybook. Please review all pages.

## Accessibility concerns or updates

- N/A

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: src/components/MultiSelect/MultiSelect.test.tsx [hunk 1]
```diff
@@ -78,7 +78,7 @@ const MultiSelectTestComponent = ({
   );
 };
 
-describe("MultiSelect Accessibility", () => {
+describe.skip("MultiSelect Accessibility", () => {
```
Comment (EdwinGuzman) [inline src/components/MultiSelect/MultiSelect.test.tsx:81]: Edwin: review.

File: .storybook/preview.tsx [hunk 2]
```diff
@@ -58,25 +57,6 @@ const customViewports = {
 const MyDocsContainer = (props) => (
   <MDXProvider
     components={{
-      h1: ({ children }) => (
```
Comment (EdwinGuzman) [inline .storybook/preview.tsx:61]: With the update in `.storybook/preview-body.html`, we can remove the DS overrides. They were mostly for styling but it previously did not allow us to use the id hash linking from headings. This is now done automatically by Storybook in their links and headings and the update in `.storybook/preview-body.html` just updates the font family for consistent styling.

File: src/components/Logo/LogoSvgs.tsx [hunk 3]
```diff
@@ -1,58 +1,58 @@
...
-import { ReactComponent as schomburgColor } from "../../../icons/svg/logo-schomburg-color.svg";
-import { ReactComponent as schomburgWhite } from "../../../icons/svg/logo-schomburg-white.svg";
-import { ReactComponent as simplyeBlack } from "../../../icons/svg/logo-simplye-black.svg";
-import { ReactComponent as simplyeWhite } from "../../../icons/svg/logo-simplye-white.svg";
-import { ReactComponent as simplyeColor } from "../../../icons/svg/logo-simplye-color.svg";
-import { ReactComponent as snflBlack } from "../../../icons/svg/logo-snfl-black.svg";
-import { ReactComponent as snflWhite } from "../../../icons/svg/logo-snfl-white.svg";
-import { ReactComponent as treasuresColor } from "../../../icons/svg/logo-treasures-color.svg";
-import { ReactComponent as treasuresColorNegative } from "../../../icons/svg/logo-treasures-color-negative.svg";
+import appleAppStoreBlack from "../../../icons/svg/logo-apple-app-store-black.svg";
```
Comment (EdwinGuzman) [inline src/components/Logo/LogoSvgs.tsx:2]: Now possible with the updated `vite-plugin-svgr` package.

File: package.json [hunk 4]
```diff
@@ -58,14 +58,11 @@
...
     "@emotion/styled": "11.11.0",
-    "downshift": "6.1.7",
     "framer-motion": "10.16.4",
     "react-datepicker": "4.14.1",
     "react-intersection-observer": "9.2.2",
     "system-font-css": "2.0.2",
-    "tough-cookie": "4.0.0",
-    "typescript": "4.9.5",
-    "vite-plugin-svgr": "2.4.0"
+    "typescript": "4.9.5"
```
Comment (bigfishdesign13) [inline package.json:65]: Why is this included here and in `peerDependencies`?

File: src/components/Accordion/Accordion.stories.tsx [hunk 5]
```diff
@@ -1,6 +1,5 @@
 import { VStack } from "@chakra-ui/react";
 import type { Meta, StoryObj } from "@storybook/react";
-import { withDesign } from "storybook-addon-designs";
```
Comment (bigfishdesign13) [inline src/components/Accordion/Accordion.stories.tsx:3]: By including this package as a dependency, it no longer needs to be explicitly imported into each story file?

File: src/components/StyleGuide/Typography.mdx [hunk 6]
```diff
@@ -521,7 +521,7 @@ New typographic styles were introduced in June 2023, adding `subtitle1`,
 `subtitle2`, `overline1` and `overline2`. As part of that update, the `tag` and
 `mini` styles have been deprecated.
 
-#### Recommedned Options
+#### Recommended Options
```
Comment (bigfishdesign13) [inline src/components/StyleGuide/Typography.mdx:524]: :)

File: .storybook/preview-body.html [hunk 7]
```diff
@@ -3,30 +3,36 @@
 
 <style>
   /* Custom heading styles */
-  .sbdocs h1.storybook-heading:not(.innerZoomElementWrapper h1) {
+  .sbdocs h1:not(.chakra-heading.innerZoomElementWrapper h1) {
```
Comment (bigfishdesign13) [inline .storybook/preview-body.html]: The `not` statement needs to be updated.  These styles are now impacting the styles of the DS `Heading` components used on the story pages.  Take a look at the `Heading` component story page.

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
I did a thorough spot check and overall it looks good.  However, the heading styles need some work.  I also had a few minor questions.

File: package.json (hunk 4 above)
Comment (EdwinGuzman) [reply package.json:65]: Good point, this should actually go in "devDependencies". We shouldn't require apps that don't use typescript to import typescript.

File: src/components/Accordion/Accordion.stories.tsx (hunk 5 above)
Comment (EdwinGuzman) [reply src/components/Accordion/Accordion.stories.tsx:3]: That's sort of true. The new stable npm package is now an official plugin so it's added in `.storybook/main.ts` as a global configuration.

File: package.json (hunk 4 above)
Comment (EdwinGuzman) [reply package.json:65]: Leaving this here and in `peerDependencies`. We need it for the Storybook build but not the React library build.

**Comment by EdwinGuzman** [comment]
@bigfishdesign13 @7emansell @oliviawongnyc @jackiequach please review this soon if possible, thanks!

**Review by 7emansell — APPROVED** [review]
Lgtm from the storybook pages i spot checked

File: src/components/Hero/Hero.stories.tsx
Comment (jackiequach) [inline src/components/Hero/Hero.stories.tsx:1]: Not sure what the issue is, but when on the Storybook page for Hero there are a bunch of errors 
image

**Review by jackiequach — CHANGES_REQUESTED** [review]
Chakra Exports > Layout component pages have not been updated to use the `of` prop.

I think these are unrelated:
- Checkbox: some table of contents items are not linking to anything
- SearchBar: 
   - component props descriptions are not shown
   - [Autocomplete SearchBar](https://nypl-design-system-git-dsd-1766-npm-package-update-nypl.vercel.app/?path=%2Fdocs%2Fcomponents-form-elements-searchbar--docs) is misaligned 
- StructuredContent: component props not shown
- I'm seeing some slowness (locally) when going into doc pages, which seems to stem from the tests for the component running on load.

**Comment by EdwinGuzman** [comment]
@jackiequach thanks! Good catch on all the issues. Some were things we missed, most likely in the react 18/charka 2.8 upgrade. Everything you flagged should be fixed except for:

> I'm seeing some slowness (locally) when going into doc pages, which seems to stem from the tests for the component running on load.

Is it all doc pages or some?

**Comment by EdwinGuzman** [comment]
Also, the autocomplete `SearchBar` example has not been working for a very long time. We'll have to fix that some time soon but not right now.

**Review by jackiequach — APPROVED** [review]
The changes look good. The slowness was across all components when I tested locally.

**Comment by EdwinGuzman** [comment]
> The changes look good. The slowness was across all components when I tested locally.

Yes, and it was related to the local test json file. It was so slow when I initially tested and it was quite annoying. I did not really change anything and then it was running faster for me.

File: src/components/Checkbox/Checkbox.mdx [hunk 8]
```diff
@@ -21,11 +21,7 @@ import { changelogData } from "./checkboxChangelogData";
 - {<Link href="#accessibility" target="_self">Accessibility</Link>}
 - {<Link href="#checked" target="_self">Checked</Link>}
 - {<Link href="#indeterminate" target="_self">Indeterminate</Link>}
-- {<Link href="#focused" target="_self">Focused</Link>}
-- {<Link href="#errored" target="_self">Errored</Link>}
-- {<Link href="#disabled" target="_self">Disabled</Link>}
-- {<Link href="#helper-text" target="_self">Helper Text</Link>}
-- {<Link href="#error-text" target="_self">Error Text</Link>}
```
Comment (bigfishdesign13) [inline src/components/Checkbox/Checkbox.mdx:28]: Good call replacing this!

File: src/docs/Chakra.mdx [hunk 9]
```diff
@@ -150,13 +150,13 @@ in the `/src/index.ts` file. Note that not _all_ Chakra components are re-export
 because we want to limit what developers should use in consuming applications.
 They include:
 
-- `Box`, `Center`, `Circle`, `Grid`, `GridItem`, `HStack`, `Square`, `Stack`, `VStack`
+`Box`, `Center`, `Circle`, `Grid`, `GridItem`, `HStack`, `Square`, `Stack`, `VStack`
 
-Note: These components are harder to have snapshot tests for because, internally,
+**Note**: These components are harder to have snapshot tests for because, internally,
```
Comment (bigfishdesign13) [inline src/docs/Chakra.mdx:155]: Let's make sure to create a ticket to start replacing these notes and warnings with the `Banner` component.

**Review by bigfishdesign13 — APPROVED** [review]
Looks good!

**Pull request merged**